From: Jean Delvare <jdelvare@suse.de>
Subject: Fix struct fw_device kABI breakage
Patch-mainline: Never
References: bnc#881697

Move the newly added field by
firewire-03-dont-use-prepare_delayed_work.patch to the tail and
protect from kABI check.

Signed-off-by: Jean Delvare <jdelvare@suse.de>
---
 include/linux/firewire.h |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/include/linux/firewire.h
+++ b/include/linux/firewire.h
@@ -200,9 +200,12 @@ struct fw_device {
 	unsigned irmc:1;
 	unsigned bc_implemented:2;
 
-	work_func_t workfn;
 	struct delayed_work work;
 	struct fw_attribute_group attribute_group;
+
+#ifndef __GENKSYMS__
+	work_func_t workfn;
+#endif
 };
 
 static inline struct fw_device *fw_device(struct device *dev)
